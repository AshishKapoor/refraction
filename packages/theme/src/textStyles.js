/**
 * Copyright (c) Quid, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
// @flow
import css from '@emotion/css/macro';
import wf from './withFallback';

const primaryFontFamily = css`
  font-family: IBM Plex Sans, Lucida Grande, Tahoma, Verdana, Arial, sans-serif;
`;

const secondaryFontFamily = css`
  font-family: IBM Plex Serif, Gudea, Lucida Grande, Tahoma, Verdana, Arial,
    sans-serif;
`;

const styles = {
  xlarge: css`
    font-size: 20px;
    line-height: 1.4;
  `,
  large: css`
    font-size: 16px;
    line-height: 1.5;
  `,
  caption: css`
    font-size: 12px;
    line-height: 1.5;
  `,
  title: css`
    font-size: 16px;
    line-height: 1.5;
    ${secondaryFontFamily}
  `,
  body: css`
    line-height: 1.57;
    ${secondaryFontFamily}
  `,
  normal: css`
    font-size: 14px;
    line-height: 1.57;
  `,
  bold: css`
    font-weight: bold;
  `,
  regular: css`
    font-weight: normal;
  `,
  secondary: wf(
    props => css`
      color: ${props.theme.secondary};
    `
  ),
  disabled: wf(
    props => css`
      color: ${props.theme.disabled};
    `
  ),
  link: wf(
    props => css`
      color: ${props.theme.link};
      cursor: pointer;
    `
  ),
  highlighted: wf(
    props => css`
      color: ${props.theme.colors.highlighted};
    `
  ),
};

const textStyles = (...ss: Array<string>) => (props: Object) => {
  const rules = ss.map(s =>
    typeof styles[s] === 'function' ? styles[s](props) : styles[s]
  );
  return css`
    ${primaryFontFamily}
    ${rules}
  `;
};

export default textStyles;
