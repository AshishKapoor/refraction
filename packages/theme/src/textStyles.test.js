/**
 * Copyright (c) Quid, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
// @flow
import * as React from 'react';
import styled from '@emotion/styled/macro';
import { mount } from 'enzyme';
import textStyles from './textStyles';

// FIXME(fzivolo): right now I can't find a way to disable source maps in these
// snapshots, once we find a way, get rid of the `.split` call

it('returns a valid CSS', () => {
  const T = styled.div`
    ${textStyles('normal')({ theme: {} })}
  `;
  expect(mount(<T />)).toMatchSnapshot();
});

it('works with "withFallback" powered styles', () => {
  const T = styled.div`
    ${textStyles('secondary', 'disabled', 'link', 'highlighted', 'body')}
  `;
  expect(mount(<T />)).toMatchSnapshot();
});
