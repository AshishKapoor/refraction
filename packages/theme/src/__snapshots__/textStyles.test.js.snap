// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns a valid CSS 1`] = `
.emotion-0 {
  font-family: IBM Plex Sans,Lucida Grande,Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 1.57;
}

<T>
  <div
    className="emotion-0 emotion-1"
  />
</T>
`;

exports[`works with "withFallback" powered styles 1`] = `
.emotion-0 {
  font-family: IBM Plex Sans,Lucida Grande,Tahoma,Verdana,Arial,sans-serif;
  color: #6C7983;
  color: #8F9BA3;
  color: #28beb8;
  color: rgba(0,193,187,0.6);
}

<T>
  <div
    className="emotion-0 emotion-1"
  />
</T>
`;
